crates/cast/src/starknet_commands/account/deploy.rs

```diff
@@ -50,27 +50,33 @@ pub async fn deploy(
     if items[&network_name][&name].is_null() {
         bail!("Account with name {name} does not exist")
     }
+    let account = &items[&network_name][&name];
 
     let private_key = SigningKey::from_secret_scalar(
         parse_number(
-            items
-                .get(&network_name)
-                .and_then(|network| network.get(&name))
-                .and_then(|name| name.get("private_key"))
+            account
+                .get("private_key")
                 .and_then(serde_json::Value::as_str)
                 .ok_or_else(|| anyhow!("Couldn't get private key from accounts file"))?,
         )
         .context("Couldn't parse private key")?,
     );
 
-    let oz_class_hash: &str = if let Some(value) = &class_hash {
-        value
-    } else {
-        OZ_CLASS_HASH
+    let oz_class_hash = {
+        if let Some(class_hash_) = &class_hash {
+            class_hash_.as_str()
+        } else if let Some(class_hash_) = account
+            .get("class_hash")
+            .and_then(serde_json::Value::as_str)
+        {
+            class_hash_
+        } else {
+            OZ_CLASS_HASH
+        }
```

Comment:
it would be cool to have some tests for this :+1:

Reply:
I've tried to add some tests but didn't manage to since we use devnet.
The only test I think could be viable here would to be check if `account deploy` fails with the `undeclared class hash` error, if some arbitrary class hash is passed.

Reply:
technically we could declare different OZ contract and deploy from it, but I can see that this is maybe an overkill for what we want to test. 
>The only test I think could be viable here would to be check if account deploy fails with the undeclared class hash error, if some arbitrary class hash is passed.

maybe this is a way to go then